ui/src/app/components/popups.tsx

```diff
@@ -26,11 +26,15 @@ const styles = {
   }
 };
 
+interface WithDynamicPositionProps {
```

Comment:
These changes helped me with debugging the popup trigger issue... having non-Any props makes it much easier to follow what's going on, and the named class helps improve stack traces.

Reply:
I definitely agree with using an interface over `any` but I'm curious where the prop's interface name shows up in stack traces.

At some point, we had made a collective decision to name Props/State interfaces within a file to just `Prop` and `State` since their scope can be inferred from the file it resides in.

But it looks like there may be a reason to reconsider?

Reply:
The props rename didn't have anything to do with stack traces... I just wanted to leave some breadcrumb that this component actually expects a 'target' property. And the reason I didn't name it "Props" is that this file has a bunch of exported components, so the next time someone tries to create a typed props interface, they'd end up colliding in name.

(The 'target' prop was especially confusing to me, because 'target' is used elsewhere in this file with an entirely different meaning and type. TBH I found the code style in this file frustrating, because it's all super loosely-typed, has zero comments and no unit tests. In an ideal world our core components would have the most tests, type-safety, and well-documented interfaces – but we sort of have the inverse here :\)